### DISCUSSION TITLE
Can caption styles for figures and tables be set differently?

### DISCUSSION THREAD

**Discussion by jacobdum** [quarto-dev/quarto-cli#9528] [Q&A]
### Description

When rendering a flextable to .docx format using quarto, captions inserted via chunk options have the style "Image Caption": 

![Screenshot 2024-04-29 at 4 12 06 PM](https://github.com/quarto-dev/quarto-cli/assets/135846021/ace7eb80-c044-4da1-b381-2d22a3eb7166)

Is it possible to have options that allow for differentiation of the associated caption (i.e., caption style for figures vs. caption style for tables)? I currently write reports using quarto rendering to .docx and have lua filters that insert and auto-populate list of figures and list of tables fields in the resultant .docx files based on caption style. However, due to this unified captioning style between figures and tables, all of the table and figure captions load under the same list as opposed to two separate lists. 

**Reproducible Example**
````qmd

---
title: "quarto flextable"
format: docx
---

# Cross-referencing

Trying out @tbl-test

```{r}
#| label: tbl-test
#| tbl-cap: Test Caption
#| echo: false
#| message: false

library(flextable)

flextable(head(mtcars))
```

````
![Screenshot 2024-04-29 at 4 52 07 PM](https://github.com/quarto-dev/quarto-cli/assets/135846021/f70c5d6e-940b-421f-a3d5-3f9ee0053b42)

**Expected Result**
 Options to differentiate cross-references between table environments and figure environments via chunk options or lua filters? 

**Session Info**
````
R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.6 LTS
flextable_0.9.4
quarto v1.4.553
````

**Comment by cscheid** [comment]
> I currently write reports using quarto rendering to .docx and have lua filters that insert and auto-populate list of figures and list of tables fields in the resultant .docx files based on caption style.

The right solution is for you to target the FloatRefTarget node, inspect the `type` field explicitly, and dispatch accordingly. 
You shouldn't need (or want) to change caption styles for that to work.

Here's an example that exercises the code path you want. Of course, instead of inserting content on the caption, you'll want to test the `float.type` element and branch according to what you need.

## `example.qmd`

````
---
title: Test Lua FloatRefTarget filter
filters:
  - findfloatreftarget.lua
---

::: {#fig-1}

A figure.

A caption for a figure.

:::


::: {#tbl-1}

A table.

A caption for a table.

:::

See @fig-1 and @tbl-1.
````

## `findfloatreftarget.lua`

```lua
function FloatRefTarget(float)
  float.caption_long.content:insert(pandoc.Str(" [This is a " .. float.type .. "]"))
  return float
end
```

**Reply by jacobdum** [reply]
I believe I understand the workflow you have outlined, but I'm not understanding how it is paired to ToC fields in MS Word? 

To my understanding, ToC fields in MS Word require explicit entries (headings, styles, bookmarks, etc.) to generate. How would targeting the float.type and dispatching appropriately relate, unless you're implying that further pre-processing of the floatreftarget is required?

Apologies if this is simpler than I'm making it out to be - quarto/lua isn't my strong suite.

**Reply by cscheid** [reply]
I don't know enough about how Word generates its ToCs, but I was responding to this:

> I currently write reports using quarto rendering to .docx and have lua filters that insert and auto-populate list of figures and list of tables fields in the resultant .docx files based on caption style

Specifically, I read this to mean that you have lua filters that operate on caption style, and I was suggesting that you instead handle the FloatRefTarget elements directly. But if you're not writing the code that collects the captions, then you'll need to learn quite a bit of internal details yourself, specifically how to emit the right OpenXML from different FloatRefTargets so that they have different caption styles. Our docx renderer is [here](https://github.com/quarto-dev/quarto-cli/blob/fa44015148e55de7114f4f64e22437e99b193cde/src/resources/filters/customnodes/floatreftarget.lua#L638-L705). You'll need to implement your own.

What you're trying to do, however, is an advanced Quarto task, so you will need some working Lua knowledge.
